src/buffer_manager.ts

```diff
@@ -722,24 +728,27 @@ export class BufferManager implements Disposable {
             // !Another hack is to retrieve cursor with delay - when we receive an external buffer the cursor pos is not immediately available
             // [1, 0]
             setTimeout(async () => {
-                const neovimCursor: [number, number] = await this.client.request("nvim_win_get_cursor", [closeWinId]);
-                if (neovimCursor) {
-                    logger.debug(
-                        `Adjusting cursor pos for external buffer: ${id}, originalPos: [${neovimCursor[0]}, ${neovimCursor[1]}]`,
-                    );
-                    const finalLine = neovimCursor[0] - 1;
-                    let finalCol = neovimCursor[1];
-                    try {
-                        finalCol = convertByteNumToCharNum(doc.lineAt(finalLine).text, neovimCursor[1]);
-                        logger.debug(`Adjusted cursor: [${finalLine}, ${finalCol}]`);
-                    } catch (e) {
-                        logger.warn(`Unable to get cursor pos for external buffer: ${id}`);
-                    }
-
-                    const selection = new Selection(finalLine, finalCol, finalLine, finalCol);
-                    editor.selections = [selection];
-                    editor.revealRange(selection, TextEditorRevealType.AtTop);
+                // Return type is known from vimdoc
+                // https://neovim.io/doc/user/api.html#nvim_win_get_cursor()
+                const neovimCursor: [number, number] = await (this.client.request("nvim_win_get_cursor", [
+                    closeWinId,
+                ]) as Promise<[number, number]>);
+
+                logger.debug(
+                    `Adjusting cursor pos for external buffer: ${id}, originalPos: [${neovimCursor[0]}, ${neovimCursor[1]}]`,
+                );
+                const finalLine = neovimCursor[0] - 1;
+                let finalCol = neovimCursor[1];
+                try {
+                    finalCol = convertByteNumToCharNum(doc.lineAt(finalLine).text, neovimCursor[1]);
+                    logger.debug(`Adjusted cursor: [${finalLine}, ${finalCol}]`);
+                } catch (e) {
+                    logger.warn(`Unable to get cursor pos for external buffer: ${id}`);
                 }
+
+                const selection = new Selection(finalLine, finalCol, finalLine, finalCol);
```

Comment:
if `convertByteNumToCharNum` why do we still set a selection? seems like that would just result in nonsense behavior

Reply:
A community contributor wrote this part, and nobody really pays much attention to external-buffer.